### PR TITLE
Unique exports

### PR DESCRIPTION
**Pull request by AustinCarr**
Fixes issue #1154 

Skips exporting "instruction" input types.

Creates a new function that creates an array of unique identifiers by using the line number and position of each field.

Adds the unique identifier to the headers.

### CODE REVIEW

**Comment by AustinCarr** [comment]
The "#get headings from non-field-based" comment seems like it's incorrect.

What's the difference between a table cell and a transcription field?
Do non-field-based transcriptions have table cells?

**Comment by benwbrum** [comment]
Table cells and transcription fields are the same thing.  We originally created the table export (and the infrastructure underlying field-based transcription) for tables embedded within textual documents, like lists of specimens collected within a naturalists' field notes, or lists of shipments of bacon to a plantation in agricultural journals.

The field-based interface simply creates a table on a page that has a single row, and suppresses the ability to add narative text around it.

File: app/controllers/export_controller.rb [hunk 1]
```diff
@@ -185,27 +185,43 @@ def update_contentdm_credentials
...
+  def get_unique_identifiers(fields)
+    uniqueIdentifiers = Array.new(fields.length)
+
+    for i in 0..(fields.length - 1) do
+      uniqueIdentifiers[i] = "#{fields[i].line_number}.#{fields[i].position}"
+    end
+
+    uniqueIdentifiers
+  end
+
```
Comment (benwbrum) [inline app/controllers/export_controller.rb]: No need to change this before we deploy, but a the Ruby `map` function can do some lovely magic here:
```ruby
unique_identifiers = fields.map { |field| "#{field.line_number}.#{field.position}"}
```

**Comment by benwbrum** [comment]
The code review is done, but we need to test this code against the Graves Diaries, Stagville Accounts, and either ADAH or IARA before we deploy to make sure that data prior to the line_number work still exports correctly.

**Comment by benwbrum** [comment]
I've tested this, and it works just fine for text+table exports.  It does change the column headers for field-based collections in ways that might be surprising to people not using the page feature.

What would you say to conditionally adding the suffix in unique_identifiers, so that a field with a unique name did not acquire the suffix, but one that had a duplicate field name would get the duplicate?  (I can implement this, if you like, but want to run the idea past you.)  That would be less likely to surprise other users.

**Comment by AustinCarr** [comment]
I think that's a great idea!

The main goal was just to allow users to name the fields whatever they want and not have to worry about column name collisions at all.

This was the most straight-forward solution I could think of.

**Comment by benwbrum** [comment]
Replaced with #1230

**Pull request closed without merging**